alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/test/java/org/alfresco/transformer/AIOControllerImageMagickTest.java

```diff
@@ -0,0 +1,151 @@
+    AIOTransformRegistry transformRegistry;
+
+    @PostConstruct
+    private void init() throws Exception
+    {
+        adapter = new ImageMagickAdapter(EXE, DYN, ROOT);
+    }
+
+    @Before @Override
+    public void before() throws IOException
+    {       
+        ReflectionTestUtils.setField(commandExecutor, "transformCommand", mockTransformCommand);
+        ReflectionTestUtils.setField(commandExecutor, "checkCommand", mockCheckCommand);
+        ReflectionTestUtils.setField(adapter, "commandExecutor", commandExecutor);
+        //Need to wire in the mocked adapter into the controller...
+        if (ReflectionTestUtils.getField(transformRegistry,"transformerTransformMapping") instanceof Map)
+        {
+            Map<String,Transformer> transformers = transformRegistry.getTransformerTransformMapping();
+            transformers.replace("imagemagick", adapter);
+            ReflectionTestUtils.setField(transformRegistry, "transformerTransformMapping", transformers);
+        }
+
+        mockTransformCommand("jpg", "png", "image/jpeg", true);
+    }
+
+    @Override
+    protected AbstractTransformerController getController() 
+    {
+        return controller;
+    }
+
+    @Override
+    protected MockHttpServletRequestBuilder mockMvcRequest(String url, MockMultipartFile sourceFile,
+        String... params)
+    {
+        final MockHttpServletRequestBuilder builder = super.mockMvcRequest(url, sourceFile, params)
+            .param("targetMimetype", targetMimetype)
+            .param("sourceMimetype", sourceMimetype);
+
+        return builder;
+    }
+
+    @Test
+    @Override
+    public void noTargetFileTest()
+    {
+        // Ignore the test in super class as the AIO transforms will not be selected .
+        // It is the mock that returns a zero length file for other transformers, when we supply an invalid targetExtension.
+    }
+   
+    @Test
+    @Override
+    public void testGetTransformConfigInfo()
+    {
+        // Ignore the test in super class as the way the AIO transformer provides config is fundementally different.
+
+
+    }
+    @Test
+    @Override
+    public void testGetInfoFromConfigWithDuplicates()
+    {
+        // Ignore the test in super class as the way the AIO transformer provides config is fundementally different.
+
+    }
+
+    @Test
+    @Override
+    public void testGetInfoFromConfigWithEmptyTransformOptions()
+    {
+        // Ignore the test in super class as the way the AIO transformer provides config is fundementally different.
+
+    }
+    
+    @Test
+    @Override
+    public void testGetInfoFromConfigWithNoTransformOptions()
+    {
+        // Ignore the test in super class as the way the AIO transformer provides config is fundementally different.
+
+    }
+    
+    @Test
+    @Override
+    public void deprecatedCommandOptionsTest() throws Exception
+    {
+        // Ignore test in super class as the CommandOptions is not within the imagemagick_engine_config.json

```

Comment:
This was added recently as part of https://github.com/Alfresco/alfresco-transform-core/commit/57e4ea4381ae016c453106bd79f53be003977624

Reply:
Override has now been removed and inherited test is run.